lib/sysrandom.rb

```diff
@@ -0,0 +1,26 @@
+require "sysrandom/version"
+
+# Secure random number generation using system RNG facilities
+module Sysrandom
+  module_function
+
+  if defined?(JRUBY_VERSION)
+    require "java"
+
+    if java.security.SecureRandom.respond_to?(:getInstanceStrong)
+      @_java_secure_random = java.security.SecureRandom.getInstanceStrong
+    else
+      @_java_secure_random = java.security.SecureRandom.getInstance("SHA1PRNG")
```

Comment:
This sort of fallback probably doesn't live up to the "sysrandom" contract, but `getInstanceStrong()` (which uses a NativePRNG backend) is only available on Java 8